engine/src/main/java/org/hibernate/search/bridge/spi/FieldType.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.bridge.spi;
+
+/**
+ * The type of an indexed field.
+ *
+ * @author Gunnar Morling
+ * @hsearch.experimental This contract is currently under active development and may be altered in future releases,
+ * breaking existing users.
+ */
+public enum FieldType {
```

Comment:
How widely we'll use it, not sure yet.

`STRING_MAPPED_BOOLEAN` and `...DATE` are needed for ES as boolean and date are supported types there (i.e. not only String and Number as in plain Lucene; At leas on the ES-API level, I suppose the internal document fields may be the old known ones).

Thinking about it, I probably should rename it to BOOLEAN and DATE.

It's all a bit guess-work at this point really. Yet I think that change will go more into the right direction than the original `getSortableFieldName()` one. We e.g. may expose more options by adding methods to the builder without breaking existing impls. But it's still marked as experimental, so IMO it's no problem if we refine (aka change) stuff afterwards.